### DISCUSSION TITLE
Issue with switchNetwork not working when using walletconnect

### DISCUSSION THREAD

**Discussion by siosio34** [wevm/wagmi#2922] [Q&A]
When switching networks in metamask mobile using walletconnect, if switchNetwork is called, nothing happens, but if you write the code below, the network switching screen appears.

<img width="705" alt="image" src="https://github.com/wagmi-dev/wagmi/assets/7166022/4b06fca2-8c5d-468e-a0bd-a423b3fcb824">

However, as far as I know, within the switchNetwork function, I know that the same function as the image is performed.

I would like to ask why this happens because switchNetwork should not be called, and Ethereum transactions are called directly through provider request.

**Comment by glitch-txs** [comment]
hey, MetaMask released a fixed for this on 7.4.x version, could you confirm you're in this version?

**Reply by siosio34** [reply]
@glitch-txs 
https://github.com/MetaMask/metamask-mobile/issues/6655#issuecomment-1684146102 After the link, there are many articles saying switchNetwork does not work in wagmi. Of course, when I tested it, it didn't work.

I am also attaching a link to the article I left about the issue.
https://github.com/MetaMask/metamask-mobile/issues/6655#issuecomment-1687343890

**Reply by glitch-txs** [reply]
just tested, seems a bit unstable, sometimes it will change the chain internally but not request on mobile and the request will appear when trying to send a tx

**Reply by siosio34** [reply]
@glitch-txs  
Can I consider this an issue of metamask mobile and not of walletconnect?

**Reply by glitch-txs** [reply]
yes, we'll do some QA and try to contact them, thanks! :)

**Reply by siosio34** [reply]
@glitch-txs It seems to still occur in metamask mobile versions 7.4.0 and 7.5.0. This may be a really rude question, but is WalletConnect also handling this issue?

**Reply by glitch-txs** [reply]
We helped them building a special SDK for them since MM is very particular wallet, we've made progress but there's still things to improve
